elastic/kibana
# [Defend Workflows][8.12] Unblock fleet setup when validating thousands of uninstall tokens

## Summary

To unblock Fleet setup when dealing with thousands of agent policies, the query that's used to get decrypted SOs is now performed in batches.

The size of the batch is related to [`max_clause_count`](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-settings.html), and it looks like the batch size needs to be smaller than the _half of the value_.
`max_clause_count` is at least `1024`, so going with `500` should work on even environments with scarcest resources.

With this modification, Fleet setup is successfully performed. Other issues (like errors in Uninstall Tokens table) have not arised.

**TODO:** updating/adding tests

## Testing
- you need to find out the value of `indices.query.bool.max_clause_count` on your local machine (is there an easy way?)
- for that, checkout `8.12`,
- add a lot of Agent Policies: so far we experienced the issue with 2.7k policies on one machine and 3.2k on another. here's a modified version of @dasansol92 's script for adding Agent Policies (without Defend): https://github.com/gergoabraham/kibana/commit/e6f24b2d7d8f1d43457524548cc52ad0129bce83
- restart Kibana
- you should see the `too_many_nested_clauses` error + you should not be able to open any Fleet pages in Kibana
(if not, add more policies)
- now, switch to this branch without stopping ES
- restart Kibana: you should see that fleet setup is successful + you should be able to access Fleet pages

## Checklist

Delete any items that are not applicable to this PR.
